gocse/README.md

```diff
@@ -0,0 +1,52 @@
+# Go CSFLE Example
+
+## Steps
+
+1. Clone this repository:
+
+   ```sh
+     git clone https://github.com/mongodb-university/csfle-guides.git
+     cd csfle-guides/gocse
+   ```
+
+   Work from the gocse directory for the remainder of these
+   instructions.
+
+2. Start a `mongod` instance (Enterprise version >= 4.2) running on port 27017
+
+3. Make sure you have the `master-key.txt` file in the root of your
+   execution environment. This is a 96-byte cryptographically-secure generated
+   master encryption key required to run this example project. To generate your
+   own master key or use a KMS, refer to the [CSFLE Use Case
+   Guide](https://docs.mongodb.com/drivers/security/client-side-field-level-encryption-guide/).
+
+   The settings for each supported KMS are located in
+   `kms/provider.go`. In this project we have used [godotenv](https://github.com/joho/godotenv) to
+   load our environmental variables. Feel free to replace the calls to `GetCheckedEnv` with string values if you
+   prefer.
+
+4. If `mongocryptd` is not installed in your system path, ensure you specify its
+   location in `csfle/encrypted_client.go` in the `EncryptedClient` function.
+
+5. Uncomment the initializer for hte KMS provider you would like to use in `main.go`
+
+   ```go
+   // awsKMS := kms.AWSProvider()
+   // azureKMS := kms.AzureProvider()
+   // gcpKMS := kms.GCPProvider()
+   localKMS := kms.LocalProvider(localKey)
+
+   ```
+
+6. Run the `go run -tags=cse .`. This does the following:
+
+   - Reads in the local key from `master-key.txt`
+   - Reads in configuration information for your preferred KMS.
```

Comment:
Do you think users might get confused if the execution paths aren't separated (as opposed to listed linearly)? It could be confusing to see that it reads in a local master key if you're using a cloud provider KMS.

Reply:
I've updated the readme to make it clear what happens depending on the kms chosen